src/typechecker.erl

```diff
@@ -3894,6 +3894,8 @@ handle_type_error({type_error, list_op_error, ListOp, P, Ty, _}) ->
               [ListOp, P, typelib:pp_type(Ty)]);
 handle_type_error({type_error, {map, _, _} = Map, ActualType, ExpectedType}) ->
     print_type_error("map", Map, ActualType, ExpectedType);
+handle_type_error({type_error, {map, _, _, _} = Map, ActualType, ExpectedType}) ->
+    print_type_error("map", Map, ActualType, ExpectedType);
```

Comment:
I suggest "map update" here, to distinguish between the messages for `M#{a=>b}` and `#{a=>b}`.